pkg/alibabacloud/eni/limits/limits.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package limits
+
+import (
+	"context"
+
+	openapi "github.com/cilium/cilium/pkg/alibabacloud/api"
+	ipamTypes "github.com/cilium/cilium/pkg/ipam/types"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// limits contains limits for adapter count and addresses. The mappings will be
+// updated from agent configuration at bootstrap time.
+//
+// Source: https://www.alibabacloud.com/help/doc-detail/25378.htm
+var limits = struct {
+	lock.RWMutex
+
+	m map[string]ipamTypes.Limits
+}{
+	m: map[string]ipamTypes.Limits{
+		"ecs.g7ne.large":    {Adapters: 3, IPv4: 10, IPv6: 0},
+		"ecs.g7ne.24xlarge": {Adapters: 15, IPv4: 50, IPv6: 0},
```

Comment:
There are a lot more instance type limits in the link provided. Do you plan to add them here?

Reply:
this is add for unit test, and all instance type is synced after operator started.

Reply:
Oh, I just read the comment; never mind :-)